types/react-instantsearch-core/index.d.ts

```diff
@@ -707,11 +707,23 @@ interface HighlightResultPrimitive {
   fullyHighlighted?: boolean;
 }
 
+export type InsightsClient = (method: InsightsClientMethod, payload: InsightsClientPayload) => void;
+
+export type InsightsClientMethod = 'clickedObjectIDsAfterSearch' | 'convertedObjectIDsAfterSearch';
+
+export interface InsightsClientPayload {
+    index: string;
+    queryID: string;
+    eventName: string;
+    objectIDs: string[];
+    positions?: number[];
+}
+
 export function EXPERIMENTAL_connectConfigureRelatedItems(
   Composed: React.ComponentType<any>
 ): React.ComponentClass<any>;
 export function connectQueryRules(Composed: React.ComponentType<any>): React.ComponentClass<any>;
-export function connectHitInsights(Composed: React.ComponentType<any>): React.ComponentClass<any>;
+export function connectHitInsights(insightsClient: InsightsClient): (props: Record<string, any>) => any;
```

Comment:
```suggestion
export function connectHitInsights(insightsClient: InsightsClient): (hitComponent: React.ComponentType<any>) => React.ComponentType<any>;
```

thanks for your fix! it's definitely wrong before your change, but isn't it more this?

Reply:
Also, would you be willing to define an interface `ConnectHitInsightsProvided` which defines the `hit` and `insights` properties that the react component will receive?

Reply:
@Haroenv yes that's true, I've just pushed a change for that

@gburgett I've added in `ConnectHitInsightsProvided` though have made the `insights` property optional since the `Hits` component can be used without insights as well. I'm new to this so not sure if that's the right approach?